### PR TITLE
Add TypeScript definitions

### PR DESCRIPTION
**Pull request by Pajn**
This allows the module to be used in a TypeScript project.

I could not get rid of the unrelated changes of the yarn.lock file.
If you does `yarn` in a clean master, will the yarn.lock file stay clean?
If so, please tell me your yarn version so that I can try with that one.

### CODE REVIEW

**Comment by Pajn** [comment]
If #10 is merged `willMount` will need to be added as well. If you want I can rebase this PR on top of #10 and add it.

**Comment by swyxio** [comment]
@Pajn i'm not a typescript expert but made my own thingy anyway. see if it helps https://github.com/sw-yx/react-classless

File: build/build.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ exec('rollup -c -f umd -o umd/reactions-component.min.js', {
   NODE_ENV: 'production',
 });
 
+console.log('\nPlacing TypeScript definitions ...');
+
+exec('cp src/index.d.ts .')
```
Comment (stereobooster) [inline build/build.js:40]: why bother with copying. You can place all definitions in typings directory directly. I do not see any win in colocating file and signature

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  didUpdate?(args: Args<P, S> & {prevProps: P; prevState: S}): void
+  willUnmount?(args: {state: S; props: P}): void
+  children?(args: Args<P, S>): void
+  render?(args: Args<P, S>): void
+}
+
+export default class Component<
+  P extends ComponentProps<P, S>,
+  S = any
+> extends React.Component<P, {}> {}
```
Comment (eddiemoore) [inline src/index.d.ts:28]: After testing this i've found it works better as
```ts
> extends React.Component<ComponentProps<P, S>, {}> {}
```

**Pull request closed without merging**